### PR TITLE
feat: add support for custom annotations

### PR DESCRIPTION
**Pull request by rawc0der**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description
I am interested in a use-case where custom annotations are read from the resources. Currently this chart is lacking support for custom annotations fields.
I have updated the templates to allow setting `podAnnotations` field from the values file in order to add pod level annotations for this chart.

Update `values.yaml` to insert your custom annotations:
```
kubernetes:
    podAnnotations: 
        your.domain/cusom-annotations: important-value
```

## Fixes 
This PR is addressing Issue #1083

### CODE REVIEW

File: helm/ingress-azure/templates/deployment.yaml [hunk 1]
```diff
@@ -27,6 +27,9 @@ spec:
         checksum/config: {{ print .Values | sha256sum }}
         prometheus.io/scrape: "true"
         prometheus.io/port: {{ .Values.kubernetes.httpServicePort | quote}}
+        {{- with .Values.podAnnotations }}
```
Comment (akshaysngupta) [inline helm/ingress-azure/templates/deployment.yaml:30]: can you add this field under `.Values.kubernetes`

**Comment by akshaysngupta** [comment]
Duplicate of: https://github.com/Azure/application-gateway-kubernetes-ingress/pull/1022

**Comment by rawc0der** [comment]
Ok, just a sec

File: helm/ingress-azure/templates/deployment.yaml (hunk 1 above)
Comment (rawc0der) [reply helm/ingress-azure/templates/deployment.yaml]: ok, annotations can be added from `.Values.kubernetes.podAnnotations` now

**Comment by rawc0der** [comment]
@akshaysngupta this is ready for merging.

**Review by 3quanfeng — APPROVED** [review]
Thanks for contribution

**Review by akshaysngupta — COMMENTED** [review]
Thanks for contributing!

**Pull request merged**